src/services/disclaimer.js

```diff
@@ -0,0 +1,153 @@
+goog.provide('ngeo.Disclaimer');
+
+goog.require('goog.asserts');
+goog.require('ngeo');
+goog.require('ngeo.Message');
+
+
+/**
+ * Provides methods to display any sort of messages, disclaimers, errors,
+ * etc. Requires Bootstrap library (both CSS and JS) to display the alerts
+ * properly.
+ *
+ * @constructor
+ * @extends {ngeo.Message}
+ * @ngdoc service
+ * @ngname ngeoDisclaimer
+ * @ngInject
+ */
+ngeo.Disclaimer = function() {
+
+  goog.base(this);
+
+  var container = angular.element('<div class="ngeo-disclaimer"></div>');
+  angular.element(document.body).append(container);
+
+  /**
+   * @type {angular.JQLite}
+   * @private
+   */
+  this.container_ = container;
+
+  /**
+   * Cache of messages.
+   * @type {Object.<string, angular.JQLite>}
+   * @private
+   */
+  this.messages_ = {};
+
+};
+goog.inherits(ngeo.Disclaimer, ngeo.Message);
+
+
+/**
+ * Show disclaimer message string or object or list of disclamer message
+ * strings or objects.
+ * @param {string|Array.<string>|ngeox.Message|Array.<ngeox.Message>}
+ *     object A message or list of messages as text or configuration objects.
+ * @export
+ */
+ngeo.Disclaimer.prototype.alert = function(object) {
+  this.show(object);
+};
+
+
+/**
+ * Close disclaimer message string or object or list of disclamer message
+ * strings or objects.
+ * @param {string|Array.<string>|ngeox.Message|Array.<ngeox.Message>}
+ *     object A message or list of messages as text or configuration objects.
+ * @export
+ */
+ngeo.Disclaimer.prototype.close = function(object) {
+  var msgObjects = this.getMessageObjects(object);
+  msgObjects.forEach(this.closeMessage_, this);
+};
+
+
+/**
+ * Show the message.
+ * @param {ngeox.Message} message Message.
+ * @protected
+ */
+ngeo.Disclaimer.prototype.showMessage = function(message) {
+  var type = message.type;
+  goog.asserts.assertString(type, 'Type should be set.');
+
+  var uid = this.getMessageUid_(message);
+  if (this.messages_[uid] !== undefined) {
+    return;
+  }
+
+  var classNames = ['alert', 'fade', 'alert-dismissible'];
+  switch (type) {
+    case ngeo.MessageType.ERROR:
+      classNames.push('alert-danger');
+      break;
+    case ngeo.MessageType.INFORMATION:
+      classNames.push('alert-info');
+      break;
+    case ngeo.MessageType.SUCCESS:
+      classNames.push('alert-success');
+      break;
+    case ngeo.MessageType.WARNING:
+      classNames.push('alert-warning');
+      break;
+    default:
+      break;
+  }
+
+  var el = angular.element(
+    '<div role="alert" class="' + classNames.join(' ') + '"></div>');
+  var button = angular.element(
+    '<button type="button" class="close" data-dismiss="alert" aria-label="' +
+    'Close' +  // FIXME i18n
+    '"><span aria-hidden="true">&times;</span></button>');
+  var msg = angular.element('<span />').html(message.msg);
+  el.append(button).append(msg);
```

Comment:
I thought about it too at first.  Here's my original thoughts about it.

A directive may not be injected.  I wanted to create something that I would inject from anywhere and ask it to show my messages.  A service was the ideal pick for that.

The next step for me is to create a directive in GMF that would define a container to hold the messages for the map.  I'd simply always ask the disclaimer service to show messages in which I define the target of that directive.

An alternative to all this would have been to keep the messages "as-is" in the service and don't manage the DOM.  We could then have a disclaimer directive listening to the messages added in there to display them accordingly.

Thoughts ?